### ISSUE TITLE
Error finding model in nanopolish call-methylation

### ISSUE THREAD

**Issue by josephhughes** [jts/nanopolish#139]
Dear Jared,

I'm running nanopolish call-methylation with the following command:

`~/bin/nanopolish/nanopolish call-methylation -v --progress -t 8 -r all.fa -g ref.fa -b reads.sorted.bam > methylation.tsv`

I've tried copying the models to the folder with the bam file is, I have also tried running the command in the directory with nanopolish but always get the same error message. 

```
Error: cannot find model with key r9_250bps.cpg.6mer.complement.pop2
HDF5-DIAG: Error detected in HDF5 (1.8.14) thread 139711169112064:
  #000: H5O.c line 240 in H5Oopen(): not a location
    major: Invalid arguments to routine
    minor: Inappropriate type
  #001: H5Gloc.c line 173 in H5G_loc(): invalid file ID
    major: Invalid arguments to routine
    minor: Bad value
HDF5-DIAG: Error detected in HDF5 (1.8.14) thread 139711169112064
```
Any advice would be welcome.

Cheers,
Joseph

**Comment by jts** [comment]
Hi,

I've just pushed a fix to skip the complement strand in this case. Could you try it?

Thanks,
Jared

**Comment by josephhughes** [comment]
If I use:
`~/bin/nanopolish/nanopolish call-methylation -v --progress -t 8 -r all.fa -g ref.fa -b reads.sorted.bam > methylation.tsv`

I get:
```

Error: cannot find model with key r9_250bps.cpg.6mer.complement.pop1
HDF5-DIAG: Error detected in HDF5 (1.8.14) thread 139629805897728:
  #000: H5O.c line 607 in H5Oget_info(): not a location
    major: Invalid arguments to routine
    minor: Inappropriate type
  #001: H5Gloc.c line 195 in H5G_loc(): invalid group ID
    major: Invalid arguments to routine
    minor: Bad value
```

And when I use:

`nanopolish call-methylation -v --progress -t 8 -r all.fa -g ref.fa -b reads.sorted.bam > methylation.tsv`

I get:
```
Error: cannot find model with key r9_250bps.cpg.6mer.complement.pop1
Segmentation fault
```
Could it be a path issue I am having?

**Comment by jts** [comment]
It may be a path issue; can you make sure you have the latest code and have performed a clean rebuild?

**Comment by jts** [comment]
Could you send me an example fast5 file?

**Comment by josephhughes** [comment]
Apologies for the slow response. I had actually forgotten to run `make` yesterday.
This is what I get with a clean install after deleting my original nanopilish folder; `git clone --recursive https://github.com/jts/nanopolish.git`; `make `

and then:

 `~/bin/nanopolish/nanopolish call-methylation -v --progress -t 8 -r all.fa -g ref.fa -b reads.sorted.bam > methylation.tsv`

```
Error: cannot find model with key r9_250bps.cpg.6mer.complement.pop1
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
Aborted
```

**Comment by josephhughes** [comment]
An example fast5 file.


[vgb_20170201_FNFAB45374_MN19940_sequencing_run_Toledo_DeltaMerlin_010217_3_98936_ch96_read2864_strand.fast5.zip](https://github.com/jts/nanopolish/files/995856/vgb_20170201_FNFAB45374_MN19940_sequencing_run_Toledo_DeltaMerlin_010217_3_98936_ch96_read2864_strand.fast5.zip)

**Comment by jts** [comment]
Thanks, what reference genome should I use?

**Comment by josephhughes** [comment]
Here's the reference
[ref.fa.zip](https://github.com/jts/nanopolish/files/996090/ref.fa.zip)

**Comment by josephhughes** [comment]
Not sure whether the first fast5 maps to the reference but this one does.
[vgb_20170316_FNFAB46402_MN19940_sequencing_run_hcmvrun2_20232_ch180_read429_strand.fast5.zip](https://github.com/jts/nanopolish/files/996105/vgb_20170316_FNFAB46402_MN19940_sequencing_run_hcmvrun2_20232_ch180_read429_strand.fast5.zip)

**Comment by jts** [comment]
Thanks for the test case, this should now be fixed (for real this time) in commit  7cb2211

**Comment by josephhughes** [comment]
It worked fine now. 
Thanks,
Joseph
